packages/neuron-wallet/src/block-sync-renderer/sync/indexer-cache-service.ts

```diff
@@ -158,7 +181,7 @@ export default class IndexerCacheService {
 
     for (const txWithStatus of txsWithStatus) {
       const {transaction, txStatus} = txWithStatus
-      const mappings = mappingsByTxHash.get(transaction.hash)
+      const mappings = mappingsByTxHash.get(transaction.hash!)!
```

Comment:
IMO too many assertions is prone to bug.

Reply:
There are two assertions here: 
1. make sure the `transaction.hash` is not empty.
2. make sure there is always a value for the hash in a `Map` in this case.

If the assertion failed, then it should throw errors to prevent the potential hidden bugs. Let me know if you got any other ways around this.